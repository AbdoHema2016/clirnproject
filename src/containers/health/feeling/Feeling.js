import React from 'react';
import {View, ScrollView, SafeAreaView, Text} from 'react-native';
import Style from './style';
import {connect} from 'react-redux';
import CLabel from '../../../components/cLabel/index';
import CButton from '../../../components/cButton/index';
import navigationService from '../../../Navigation/NavigationService';
import {
  feelingSelectedAction,
  sendHealthStatAction,
} from '../feeling/redux/actions';
import {signUpStepAction} from '../../onBoarding/personalDetails/redux/actions';
import AsyncStorage, {AsyncConstants} from '../../../utilities/AsyncStorage';
import CNavigationBackButton from '../../../components/cNavigationBackButton';
import {signUpSteps, testIds, analyticsIds} from '../../../utilities';

import {translate} from '../../../Localization';

import {logAnalyticsEvent} from '../../../utilities/Firebase';

class Feeling extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {};
    navigationService.navigation = this.props.navigation;
  }

  componentDidMount() {
    this.navigationOptions();
  }

  navigationOptions = () => {
    this.props.navigation.setOptions({
      title: '',
      headerTransparent: true,
      headerLeft: () => (
        <CNavigationBackButton
          title={translate('BACK.BACK_BUTTON_TITLE')}
          backButtonAction={this.goBack}
        />
      ),
    });
  };

  goBack = () => {
    this.props.navigation.navigate('VerifyDocuments');
  };

  feelingSelected = (index, value) => {
    this.props.setFeeling(value, index);
  };

  onSkipPressed = () => {
    this.props.signupStep('6');
    AsyncStorage.setItemInStorage(AsyncConstants.STEP, signUpSteps.Temperature);
    this.props.navigation.navigate('Temperature');
    logAnalyticsEvent(analyticsIds.onboard_feeling_skipped);
  };

  sendHealthStat = () => {
    const {
      tempSkipped,
      modalIndex,
      feeling: {
        feeling: {value},
      },
    } = this.props;
    let data = {
      params: {
        health_status: value,
        step: 5,
      },
      token: this.props.access_token,
      via: 'signup',
      skip: tempSkipped,
      modalIndex: modalIndex,
    };
    logAnalyticsEvent(analyticsIds.onboard_feeling_added);
    this.props.sendHealthStat(data);
  };

  render() {
    return (
      <SafeAreaView style={Style.container}>
        <ScrollView style={Style.innerContainer} bounces={false}>
          <View style={Style.lablesContainer}>
            <CLabel style={Style.infoStageLabel} text={'5/7'} />
            <CLabel
              style={Style.appTitle}
              text={'How are you feeling right now?'}
            />
            <CLabel style={Style.chooseAns} text={'Choose the answer'} />
          </View>

          <View style={Style.answerButtons}>
            <CButton
              testID={testIds.feelingOne}
              accessibilityLabel={testIds.feelingOne}
              onPress={() => this.feelingSelected(0, 'I feel great')}
              text={'I feel great'}
              textStyle={Style.blackText}
              buttonContainerStyle={
                this.props.feeling.feeling.index === 0
                  ? Style.selectedButton
                  : Style.unselectedButton
              }
              backgroundColor={Style.selectedButton.backgroundColor}
            />
            <CButton
              testID={testIds.feelingTwo}
              accessibilityLabel={testIds.feelingTwo}
              onPress={() => this.feelingSelected(1, 'I feel normal')}
              text={'I feel normal'}
              textStyle={Style.blackText}
              buttonContainerStyle={
                this.props.feeling.feeling.index === 1
                  ? Style.selectedButton
                  : Style.unselectedButton
              }
              backgroundColor={Style.selectedButton.backgroundColor}
            />
            <CButton
              testID={testIds.feelingThree}
              accessibilityLabel={testIds.feelingThree}
              onPress={() => this.feelingSelected(2, 'I’m not too sure')}
              text={'I’m not too sure'}
              textStyle={Style.blackText}
              buttonContainerStyle={
                this.props.feeling.feeling.index === 2
                  ? Style.selectedButton
                  : Style.unselectedButton
              }
              backgroundColor={Style.selectedButton.backgroundColor}
            />
            <CButton
              testID={testIds.feelingFour}
              accessibilityLabel={testIds.feelingFour}
              onPress={() => this.feelingSelected(3, 'I don’t feel well')}
              text={'I don’t feel well'}
              textStyle={Style.blackText}
              buttonContainerStyle={
                this.props.feeling.feeling.index === 3
                  ? Style.selectedButton
                  : Style.unselectedButton
              }
              backgroundColor={Style.selectedButton.backgroundColor}
            />
          </View>
        </ScrollView>
        <View style={Style.buttonsContainer}>
          <CButton
            testID={testIds.continue}
            accessibilityLabel={testIds.continue}
            disabled={this.props.feeling.feeling.value == null ? true : false}
            text={'Continue'}
            textStyle={Style.whiteText}
            loading={this.props.feeling.loading}
            onPress={() => this.sendHealthStat()}
            buttonContainerStyle={
              this.props.feeling.feeling.value == null
                ? Style.disabledButton
                : Style.abledGreenButton
            }
          />
          <Text
            accessible={true}
            testID={testIds.skip}
            accessibilityLabel={testIds.skip}
            style={Style.skip}
            onPress={() => this.onSkipPressed()}>
            {translate('STRINGS.SKIP')}
          </Text>
        </View>
      </SafeAreaView>
    );
  }
}

const mapStateToProps = ({
  feeling,
  personalDetails,
  userProfile,
  selectTempType,
}) => {
  return {
    feeling,
    access_token: personalDetails.access_token,
    tempSkipped: selectTempType.skip,
    modalIndex: userProfile.modalIndex,
  };
};

const mapDispatchToProps = {
  setFeeling: feelingSelectedAction,
  sendHealthStat: sendHealthStatAction,
  signupStep: signUpStepAction,
};

export default connect(mapStateToProps, mapDispatchToProps)(Feeling);
