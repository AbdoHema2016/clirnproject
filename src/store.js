import {createStore, applyMiddleware} from 'redux';
import createSagaMiddleware from 'redux-saga';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import {rootSaga} from './sagas';
const sagaMiddleware = createSagaMiddleware();
const isDevelopmentEnv =
  process && process.env && process.env.NODE_ENV === 'development';
const isProductionEnv =
  process && process.env && process.env.NODE_ENV === 'production';
const logStoreInfo = false;
const middlewares = [sagaMiddleware];
function actionLogger({getState}) {
  return (next) => (action) => {
    let allowed = [
      'jumio',
      'notificationsReducer',
      'otp',
      'companies',
      'signIn',
      'updateContactDetailsReducer',
      'userProfile',
      'vaccinations',
    ];
    let state = getState();
    let loggedStates = Object.keys(state)
      .filter((key) => allowed.includes(key))
      .reduce((obj, key) => {
        return {
          ...obj,
          [key]: state[key],
        };
      }, {});
    console.info({actionType: action.type, loggedStates});
    const returnValue = next(action);
    return returnValue;
  };
}
if (isDevelopmentEnv && logStoreInfo) {
  middlewares.push(createLogger());
}
if (isProductionEnv) {
  middlewares.push(actionLogger);
}

const store = createStore(rootReducer, applyMiddleware(...middlewares));
sagaMiddleware.run(rootSaga);
export default store;
