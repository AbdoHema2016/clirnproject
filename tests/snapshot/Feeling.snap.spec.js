import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import store from '../../src/store';
import Feeling from '../../src/containers/health/feeling/Feeling';

jest.mock('@react-navigation/native', () => ({
  withNavigationFocus: (component) => component,
}));

jest.useFakeTimers();

describe('Feeling screen', () => {
  const props = {
    navigation: {
      setOptions: jest.fn(),
    },
  };
  it('Snapshot Feeling', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Feeling {...props} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
