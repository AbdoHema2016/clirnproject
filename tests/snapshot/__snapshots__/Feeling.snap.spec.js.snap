// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feeling screen Snapshot Feeling 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F7F7F7",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    bounces={false}
    style={
      Object {
        "bottom": 24,
        "width": 750,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "width": 722,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#8D8C91",
                "fontFamily": "SofiaPro-Medium",
                "fontSize": 16,
                "marginTop": 160.07999999999998,
              },
            ]
          }
        >
          5/7
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#2B342C",
                "fontFamily": "SofiaPro-Medium",
                "fontSize": 24,
                "fontWeight": "600",
              },
            ]
          }
        >
          How are you feeling right now?
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#8D8C91",
                "fontFamily": "SofiaPro-Medium",
                "fontSize": 18,
                "marginTop": 26,
              },
            ]
          }
        >
          Choose the answer
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginBottom": 8,
            "marginTop": 104.052,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 4,
                "height": 52,
                "justifyContent": "center",
                "shadowColor": "#DDDDDD",
                "shadowOffset": Object {
                  "height": 8,
                  "width": 0,
                },
                "shadowOpacity": 0.75,
                "shadowRadius": 4,
              },
              Object {
                "backgroundColor": "#FFFFFF",
                "marginTop": 20,
                "width": 724,
              },
            ]
          }
        >
          <View
            accessibilityLabel="FeelingOne"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderRadius": 4,
                "height": "100%",
                "justifyContent": "center",
                "opacity": 1,
                "width": 718,
              }
            }
            testID="FeelingOne"
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontFamily": "SofiaProRegular",
                    "fontSize": 17,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#2B342C",
                  },
                ]
              }
            >
              I feel great
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 4,
                "height": 52,
                "justifyContent": "center",
                "shadowColor": "#DDDDDD",
                "shadowOffset": Object {
                  "height": 8,
                  "width": 0,
                },
                "shadowOpacity": 0.75,
                "shadowRadius": 4,
              },
              Object {
                "backgroundColor": "#FFFFFF",
                "marginTop": 20,
                "width": 724,
              },
            ]
          }
        >
          <View
            accessibilityLabel="FeelingTwo"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderRadius": 4,
                "height": "100%",
                "justifyContent": "center",
                "opacity": 1,
                "width": 718,
              }
            }
            testID="FeelingTwo"
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontFamily": "SofiaProRegular",
                    "fontSize": 17,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#2B342C",
                  },
                ]
              }
            >
              I feel normal
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 4,
                "height": 52,
                "justifyContent": "center",
                "shadowColor": "#DDDDDD",
                "shadowOffset": Object {
                  "height": 8,
                  "width": 0,
                },
                "shadowOpacity": 0.75,
                "shadowRadius": 4,
              },
              Object {
                "backgroundColor": "#FFFFFF",
                "marginTop": 20,
                "width": 724,
              },
            ]
          }
        >
          <View
            accessibilityLabel="FeelingThree"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderRadius": 4,
                "height": "100%",
                "justifyContent": "center",
                "opacity": 1,
                "width": 718,
              }
            }
            testID="FeelingThree"
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontFamily": "SofiaProRegular",
                    "fontSize": 17,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#2B342C",
                  },
                ]
              }
            >
              I’m not too sure
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 4,
                "height": 52,
                "justifyContent": "center",
                "shadowColor": "#DDDDDD",
                "shadowOffset": Object {
                  "height": 8,
                  "width": 0,
                },
                "shadowOpacity": 0.75,
                "shadowRadius": 4,
              },
              Object {
                "backgroundColor": "#FFFFFF",
                "marginTop": 20,
                "width": 724,
              },
            ]
          }
        >
          <View
            accessibilityLabel="FeelingFour"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderRadius": 4,
                "height": "100%",
                "justifyContent": "center",
                "opacity": 1,
                "width": 718,
              }
            }
            testID="FeelingFour"
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontFamily": "SofiaProRegular",
                    "fontSize": 17,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#2B342C",
                  },
                ]
              }
            >
              I don’t feel well
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "bottom": 130.732,
        "height": 50,
        "justifyContent": "space-between",
        "width": 722,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 4,
            "height": 52,
            "justifyContent": "center",
            "shadowColor": "#DDDDDD",
            "shadowOffset": Object {
              "height": 8,
              "width": 0,
            },
            "shadowOpacity": 0.75,
            "shadowRadius": 4,
          },
          Object {
            "backgroundColor": "#8D8C91",
          },
        ]
      }
    >
      <View
        accessibilityLabel="Continue"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": undefined,
            "borderRadius": 4,
            "height": "100%",
            "justifyContent": "center",
            "opacity": 1,
            "width": 718,
          }
        }
        testID="Continue"
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontFamily": "SofiaProRegular",
                "fontSize": 17,
                "textAlign": "center",
              },
              Object {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          Continue
        </Text>
      </View>
    </View>
    <Text
      accessibilityLabel="Skip"
      accessible={true}
      onPress={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "color": "#2B342C",
          "fontFamily": "SofiaProRegular",
          "fontSize": 16,
          "marginTop": 5,
          "paddingVertical": 15,
          "textAlign": "center",
          "width": 694,
        }
      }
      testID="Skip"
    >
      [missing "en.STRINGS.SKIP" translation]
    </Text>
  </View>
</View>
`;
